src/compile/axis.ts

```diff
@@ -61,6 +61,19 @@ export function parseInnerAxis(channel: Channel, model: Model): VgAxis {
     }
   });
 
+  const props = model.axis(channel).properties || {};
+
+  // it might have more to be included here.
+  ['grid'].forEach(function(group) {
+    const value = properties[group] ?
```

Comment:
Good question.   I would examine properties of `'axis', 'labels', 'title', 'ticks', 'majorTicks', 'minorTicks'`  and see if they have similar problem.